app/kumactl/pkg/tokens/zoneegress_client.go

```diff
@@ -0,0 +1,66 @@
+package tokens
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+
+	error_types "github.com/kumahq/kuma/pkg/core/rest/errors/types"
+	"github.com/kumahq/kuma/pkg/tokens/builtin/server/types"
+	util_http "github.com/kumahq/kuma/pkg/util/http"
+)
+
+func NewZoneEgressTokenClient(client util_http.Client) ZoneEgressTokenClient {
+	return &httpZoneEgressTokenClient{
+		client: client,
+	}
+}
+
+type ZoneEgressTokenClient interface {
+	Generate(zone string, validFor time.Duration) (string, error)
+}
+
+type httpZoneEgressTokenClient struct {
+	client util_http.Client
+}
+
+var _ ZoneEgressTokenClient = &httpZoneEgressTokenClient{}
+
+func (h *httpZoneEgressTokenClient) Generate(zone string, validFor time.Duration) (string, error) {
+	tokenReq := &types.ZoneEgressTokenRequest{
+		Zone:     zone,
+		ValidFor: validFor.String(),
+	}
+	reqBytes, err := json.Marshal(tokenReq)
+	if err != nil {
+		return "", errors.Wrap(err, "could not marshal token request to json")
+	}
+	req, err := http.NewRequest("POST", "/tokens/zone-egress", bytes.NewReader(reqBytes))
+	if err != nil {
+		return "", errors.Wrap(err, "could not construct the request")
+	}
+	req.Header.Set("content-type", "application/json")
+	resp, err := h.client.Do(req)
```

Comment:
Might be better just to use `http.Post`, it's very noisy otherwise and I waste time looking for the "non-typical" manipulation of the request when I see `NewRequest` + `Do`

Reply:
Is there really no abstraction that lets us avoid copying `Generate`?

Reply:
It's using custom http client under she scene, so I think there is no cleaner way of doing it, but I may be wrong and am definitely open to suggestions

Reply:
I think you can do `h.client.Post(..)` then

Reply:
I just checked, and we created an abstraction over `http.Client` so I may add additional member functions there and we may do it as you suggest